### DISCUSSION TITLE
How to give color and percentages In Neper?

### DISCUSSION THREAD

**Discussion by Ali141272** [neperfepx/neper#579] [Q&A] [answered]
Hello everyone,
How to give a specific color to different minerals. Can I also give a percentage to each mineral in Neper? An example is shown in the attachment.
Thank you. 
![Capture1](https://user-images.githubusercontent.com/102291665/194884719-fab580a5-c0d8-454d-9a41-2ebfea466409.PNG)

**Comment by rquey** [comment, marked as answer]
The most direct way is to define groups (and colors) from a file. For example, for 1000 grains:

```
neper -T -n 1000 -domain "square(1,2)" -morpho "diameq:lognormal(1,0.1),1-sphericity:lognormal(0.1,0.03)" -dim 2 -morphooptistop "val=1e-1"
echo | awk '{for (i=0; i<1000; i++)
                 if (i < 650) {print "1"}
            else if (i < 800) {print "2"}
            else if (i < 900) {print "3"}
            else if (i < 950) {print "4"}
            else              {print "5"}}' > group
echo | awk '{for (i=0; i<1000; i++)
                 if (i < 650) {print "red"}
            else if (i < 800) {print "orange"}
            else if (i < 900) {print "yellow"}
            else if (i < 950) {print "lightblue"}
            else              {print "blue"}}' > color
neper -V n1000-id1.tess -datacellcol "col:file(color)" -print img
```
![image](https://user-images.githubusercontent.com/2659376/199468302-476b9a35-25ce-4088-a1c9-3e969b6ef6e2.png)

**Reply by ashokambig** [reply]
Dear Quey,

Is there a way to export these groups as elemental sets readable in abaqus (as .inp file)?
If yes, could you please send me the neper script for the above example?
I need to create an artificial mircrostructure by considering volume fractions of different minerals.
Thanks for your answer.

Regards
Ashok

**Comment by Ali141272** [comment]
Professor Dr. Romain Quey, Thank you very much for your kind response.
